src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -186,6 +214,25 @@ private async Task ProcessQueueAsync()
                     var context = await work.CreateRequestContextAsync(cancellationToken).ConfigureAwait(false);
                     if (work.MutatesServerState)
                     {
+                        if (CancelInProgressWorkUponMutatingRequest)
+                        {
+                            // Cancel all concurrently executing tasks
+                            var concurrentlyExecutingTasksArray = concurrentlyExecutingTasks.ToArray();
+                            for (var i = 0; i < concurrentlyExecutingTasksArray.Length; i++)
+                            {
+                                concurrentlyExecutingTasksArray[i].Value.Cancel();
+                            }
+
+                            try
+                            {
+                                // wait for all pending tasks to complete their cancellation
+                                await Task.WhenAll(concurrentlyExecutingTasksArray.Select(kvp => kvp.Key)).ConfigureAwait(false);
+                            }
+                            catch (TaskCanceledException)
```

Comment:
```suggestion
                            catch (OperationCanceledException)
```

(though using jason's suggestion is better) :)